waltid-cli/src/commonMain/kotlin/id/walt/cli/commands/VCVerifyCmd.kt

```diff
@@ -0,0 +1,218 @@
+        - expired: Verifies that the credentials expiration date (`exp` for JWTs) has not been exceeded.
+        - webhook: Sends the credential data to an webhook URL as HTTP POST, and returns the verified status based on the webhooks set status code (success = 200 - 299).
+        - maximum-credentials: Verifies that a maximum number of credentials in the Verifiable Presentation is not exceeded.
+        - minimum-credentials: Verifies that a minimum number of credentials are included in the Verifiable Presentation.
+        - signature: Checks a JWT credential by verifying its cryptographic signature using the key referenced by the DID in `iss`.
+        - allowed-issuer: Checks that the issuer of the credential is present in the supplied list.
+        - not-before: Verifies that the credentials not-before date (for JWT: `nbf`, if unavailable: `iat` - 1 min) is correctly exceeded.
+        
+        Multiple policies are accepted. e.g.
+     
+            waltid vc verify --policy=signature --policy=expired vc.json
+   
+        If no policy is specified, only the Signature Policy will be applied. i.e.
+        
+            waltid vc verify vc.json
+        
+        Some policies require parameters. To specify it, use --arg or -a options. e.g.
+        
+            --arg=param1=value1 --a param2=value2
+            
+            e.g.
+
+            waltid vc verify --policy=schema -a schema=mySchema.json vc.json
+        """.trimIndent()
+    }
+
+
+    val print: PrettyPrinter = PrettyPrinter(this)
+
+    private val vc: File by argument(help = "the verifiable credential file (in JWS format) to be verified (required)").file()
+
+    val policies: List<String> by option(
+        "-p",
+        "--policy",
+        help = """Specify a policy to be applied in the verification process."""
+    ).choice(
+        "schema",
+        "holder-binding",
+        "expired",
+        "webhook",
+        "maximum-credentials",
+        "minimum-credentials",
+        "signature",
+        "allowed-issuer",
+        "not-before"
+    ).multiple()
+
+    val policyArguments: Map<String, String> by option(
+        "-a",
+        "--arg",
+    ).associate().help {
+        """Argument required by some policies, namely:
+            
+            |Policy|Expected Argument|
+            |------|--------|
+            |signature| - |
+            |expired| - |
+            |not-before| - |
+            |schema|schema=/path/to/schema.json|
+        """.trimMargin()
+    }
+
+    override fun run() {
+
+        val jws = vc.readText()
+        val args = checkAndLoadArguments()
+
+        val results = runBlocking { VCUtil.verify(jws, policies, args) }
+
+        print.box("Verification Result")
+        results.forEach {
+            if (it.isSuccess()) { // Not enough to be a successful verification. Sometimes, the verification succeeds because the policy is not even applied.
```

Comment:
I added this check to cover cases where the mandatory fields for this specific policy were missing. 

When I added this, however, what happened was that there was a flaw in the VC creation itself. 

Now, I'm not sure whether we should:

(a) keep this check and fail if the fields needed to apply the policy are not found
(b) keep this check, pass the verification but warning that the policy was not applied because the necessary fields were not found
(c) remove this check and just pass the check without saying that it wasn't even applied

What do you think, @waltkb and @philpotisk?

Reply:
Talked to Phil and clarified me that 

- if a VC doesn't have the required fields for expiration policy validation, it means it doesn't expire.
- if a VC doesn't have the required fields for not-before policy validation, it means it's valid from the day of issue.
- as such, if the policy is not applied because o lacking fields, it's ok for some policies (but we didn't check all)
- if we need to handle any specific case, it should be done at the VC library level.